src/main/host/thread_preload.c

```diff
@@ -142,19 +142,16 @@ static pid_t _threadpreload_fork_exec(ThreadPreload* thread, const char* file, c
             // child
 
             // Set the working directory
-            utility_assert(workingDir != NULL);
             if (chdir(workingDir) < 0) {
-                error("chdir(%s): %s", workingDir, g_strerror(errno));
+                die_after_vfork();
             }
 
             int rc = execvpe(file, argv, envp);
             if (rc == -1) {
```

Comment:
`execvpe()` isn't part of `man 2`. Since Shadow forces the executable path to be an absolute path anyways, I think we should change this to be `execve()`.

Reply:
The `exec` family of functions is explicitly whitelisted in vfork(2):

> (From  POSIX.1) The vfork() function has the same effect as fork(2), except that the behavior is undefined if the process created by vfork() either modifies any data other than a variable of type pid_t used to store the return value from vfork(), or returns from the function in which vfork() was called, or calls any other function before  successfully calling _exit(2) or one of the exec(3) family of functions.

Granted this also indicates that we're in undefined behavior land doing pretty much anything at all in between `vfork` and `exec*`, but I think they're being a bit conservative.